shell/platform/linux/fl_engine.cc

```diff
@@ -94,34 +94,42 @@ static void parse_locale(const gchar* locale,
 // Passes locale information to the Flutter engine.
 static void setup_locales(FlEngine* self) {
   const gchar* const* languages = g_get_language_names();
-  g_autoptr(GPtrArray) locales = g_ptr_array_new_with_free_func(g_free);
+  g_autoptr(GPtrArray) locales_array = g_ptr_array_new_with_free_func(g_free);
   // Helper array to take ownership of the strings passed to Flutter.
   g_autoptr(GPtrArray) locale_strings = g_ptr_array_new_with_free_func(g_free);
   for (int i = 0; languages[i] != nullptr; i++) {
     gchar *language, *territory, *codeset, *modifier;
     parse_locale(languages[i], &language, &territory, &codeset, &modifier);
-    if (language != nullptr)
+    if (language != nullptr) {
       g_ptr_array_add(locale_strings, language);
-    if (territory != nullptr)
+    }
+    if (territory != nullptr) {
       g_ptr_array_add(locale_strings, territory);
-    if (codeset != nullptr)
+    }
+    if (codeset != nullptr) {
       g_ptr_array_add(locale_strings, codeset);
-    if (modifier != nullptr)
+    }
+    if (modifier != nullptr) {
       g_ptr_array_add(locale_strings, modifier);
+    }
 
     FlutterLocale* locale =
         static_cast<FlutterLocale*>(g_malloc0(sizeof(FlutterLocale)));
-    g_ptr_array_add(locales, locale);
+    g_ptr_array_add(locales_array, locale);
     locale->struct_size = sizeof(FlutterLocale);
     locale->language_code = language;
     locale->country_code = territory;
     locale->script_code = codeset;
     locale->variant_code = modifier;
   }
+  FlutterLocale** locales =
+      reinterpret_cast<FlutterLocale**>(locales_array->pdata);
```

Comment:
Is a reinterpret cast actually needed here? Isn't this a `void*` under the hood (in which case static_cast should work)?

Reply:
It's a `void**` under the hood, which apparently doesn't work:
```
error: static_cast from 'gpointer *' (aka 'void **') to 'FlutterLocale **' is not allowed
```